### PR TITLE
DDF-3297 Admin UI permission check performance

### PR DESCRIPTION
**Pull request by pklinef**
#### What does this PR do?
Admin UI operations have slowed down with the security manager due to permission checks related to different OSGi operations such as looking up the bundle location and getting metatype information.  Caching this information improves Admin UI responsiveness. 

#### Who is reviewing it? 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
@tbatie 
@vinamartin 

#### Select relevant component teams: 
@codice/security 

#### Choose 2 committers to review/merge the PR. 
@coyotesqrl
@stustison

#### How should this be tested? (List steps with links to updated documentation)
Verify the response type to show applications in the Admin UI drops from about 2 seconds to about half a second which is close to pre-security manager response times.

#### What are the relevant tickets?
[DDF-3297](https://codice.atlassian.net/browse/DDF-3297)

#### Checklist:
- [ n/a ] Documentation Updated
- [ n/a ] Update / Add Unit Tests
- [ n/a ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceBean.java [hunk 1]
```diff
@@ -421,4 +428,8 @@ protected BundleContext getContext() {
   void setServiceTracker(ServiceTracker serviceTracker) {
     this.serviceTracker = (ServiceTracker<Object, Object>) serviceTracker;
   }
+
+  private static String computeLocation(Bundle bundle) {
```
Comment (pklinef) [inline platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceBean.java]: Bundle location caching is duplicated in the application service code but this code is deprecated for removal.

File: platform/osgi/platform-osgi-conditions/src/main/java/org/codice/ddf/condition/BundleNameCondition.java [hunk 2]
```diff
@@ -76,7 +79,7 @@ public boolean isSatisfied() {
       return false;
     }
     String bundleName = bundle.getHeaders().get("Bundle-SymbolicName");
-    String key = bundleName + args[0];
+    String key = bundleName + Arrays.toString(args);
```
Comment (stustison) [inline platform/osgi/platform-osgi-conditions/src/main/java/org/codice/ddf/condition/BundleNameCondition.java]: These changes are going to conflict a bit with my test PR

**Comment by pklinef** [comment]
build now

**Comment by pklinef** [comment]
build now

**Comment by pklinef** [comment]
build now

**Comment by vinamartin** [comment]
Manually verified that the load times are faster

**Pull request merged**